### ISSUE TITLE
Timeouts do not work

### ISSUE THREAD

**Issue by johnspackman** [browserless/chrome#3496]
**Describe the bug**
The timeouts set by `page.setDefaultNavigationTimeout` and `page.setDefaultTimeout` are not respected by browserless, but work fine when using "normal" puppeteer and chromium.

**To Reproduce**
1. Create a url that is very slow (eg has a 10 second delay)
2. Run docker: 
```
$ docker run --rm -p 3000:3000 -e "MAX_CONCURRENT_SESSIONS=10" --add-host myhostname.co.uk:host-gateway browserless/chrome:latest
```

3. Run this code:
```
const puppeteer = require("puppeteer-core");

async function test() {
  const browser = await puppeteer.connect({ browserWSEndpoint: "ws://localhost:3000" });
  const page = await browser.newPage();

  page.setDefaultNavigationTimeout(30000);
  page.setDefaultTimeout(30000);
  await page.goto("http://myhostname.co.uk/my-slow-url.html");
  const data = await page.screenshot({ path: "./x.png", fullPath: true });
  await browser.close();
}

test();
```

**Expected behavior**
It should work and not fail with a timeout.  What actually happens is that Chromium reports `net::ERR_TIMED_OUT` after only a few seconds

**Screenshots**
```
$ node mytest.js  
/home/dev/mytestapp/node_modules/puppeteer-core/lib/cjs/puppeteer/common/Frame.js:133
                    ? new Error(`${response.errorText} at ${url}`)
                      ^

Error: net::ERR_TIMED_OUT at http://myhostname.co.uk/my-slow-url.html
    at navigate (/home/dev/mytestapp/node_modules/puppeteer-core/lib/cjs/puppeteer/common/Frame.js:133:23)
    at async Deferred.race (/home/dev/mytestapp/node_modules/puppeteer-core/lib/cjs/puppeteer/util/Deferred.js:82:20)
    at async Frame.goto (/home/dev/mytestapp/node_modules/puppeteer-core/lib/cjs/puppeteer/common/Frame.js:99:21)
    at async CDPPage.goto (/home/dev/mytestapp/node_modules/puppeteer-core/lib/cjs/puppeteer/common/Page.js:668:16)
    at async test (/home/dev/mytestapp/x.js:9:3)

Node.js v20.3.0
```

**Desktop (please complete the following information):**
 - OS: Mac OS 13.4.1 Ventura
 - Browser: Chromium
 - Version: browserless/chromium:latest dcae9356daf708868586191768fcde3ebaec18b5b955e763cc24b18162f3664a

**Additional context**
This works fine if you use ordinary chromium,. eg this code:

```
const puppeteer = require("puppeteer");

async function test() {
  const browser = await puppeteer.launch();
  const page = await browser.newPage();

  page.setDefaultNavigationTimeout(0);
  page.setDefaultTimeout(0);
  await page.goto("http://myhostname.co.uk/my-slow-url.html");
  const data = await page.screenshot({ path: "./x.png", fullPath: true });
  await browser.close();
}

test();
```

works as expected

**Comment by johnspackman** [comment]
Browerless corporate sales want to charge US$2,400 per year just to use this container ... and the development team can't even respond to issues about broken software.

**Comment by andyMrtnzP** [comment]
Sorry for the delay. Your script is timing out because it's hitting the _server_ timeout, which can't be overridden by Puppeteer. You can override that timeout both at the Docker level and the single-connection level.

To set it at docker level use the `CONNECTION_TIMEOUT` flag:
`docker run --rm -p 3000:3000 -e "CONNECTION_TIMEOUT=600000" browserless/chrome:latest`

To set it at connection level, add `timeout` to the query string:
`await puppeteer.connect({ browserWSEndpoint: "ws://localhost:3000?timeout=600000" });`

**Comment by joelgriffith** [comment]
@johnspackman we do monitor issues here, but are much more responsive through the support channel (support@browserless.io). We'll make this more clear in issue templates since we hate to see things like this "fall through the cracks"

**Comment by Nurgak** [comment]
The `CONNECTION_TIMEOUT` environment variable, [renamed to `TIMEOUT` now](https://github.com/browserless/browserless/blob/main/MIGRATION-2.0.md#changed-parameters), should really be made more obvious, perhaps in the README?

For my use-case, browserless is running on a Raspberry Pi 4, obviously it's somewhat slow and my process takes many navigation steps, eventually running in the timeout issue. It took a lot of digging to understand what was happening and why (mostly due to lack of experience).

As @johnspackman said, none of the usual ways, that pop up in the search engines, to control the timeout worked (`page.setDefaultNavigationTimeout`, `page.setDefaultTimeout`). Eventually, by explicitly searching the past issues here I found this issue where the answer was provided.

Here are the logs so this thread could be found via search engines in the future:
```sh
puppeteer  |   browserless.io:limiter:warn  Job has hit timeout after 30,003ms of activity. +4m
puppeteer  |   browserless.io:limiter:info  Calling timeout handler +30s
puppeteer  |   browserless.io:router:error  Websocket job has timedout, sending 429 response +4m
puppeteer  |   browserless.io:limiter:info  (Running: 0, Pending: 0) All jobs complete.  +1ms
puppeteer  |   browserless.io:router:trace  WebSocket Request handler has finished. +25s
puppeteer  |   browserless.io:browser-manager:info  1 Client(s) are currently connected, Keep-until: 0, force: false +4m
puppeteer  |   browserless.io:server:trace  Websocket connection complete +23s
puppeteer  |   browserless.io:router:trace  WebSocket Request handler has finished. +16ms
puppeteer  |   browserless.io:browser-manager:info  0 Client(s) are currently connected, Keep-until: 0, force: false +16ms
puppeteer  |   browserless.io:browser-manager:info  Closing browser session +1ms
puppeteer  |   browserless.io:browser-manager:info  Deleting "/tmp/browserless-data-dirs/browserless-data-dir-736ba4dd-1242-4ecd-ab5c-b71b35452118" user-data-dir and session from memory +0ms
puppeteer  |   browserless.io:ChromiumCDPWebSocketRoute:info 172.19.0.1 Closing ChromiumCDP process and all listeners +25s
puppeteer  |   browserless.io:server:trace  Websocket connection complete +21ms
puppeteer  |   browserless.io:browser-manager:info  Deleting data directory "/tmp/browserless-data-dirs/browserless-data-dir-736ba4dd-1242-4ecd-ab5c-b71b35452118" +7ms
```

Basically, run docker with `-e TIMEOUT=60000` and the session will not be killed after 30s.
